packages/lisk-transactions/src/0_transfer_transaction.ts

```diff
@@ -161,20 +150,34 @@ export class TransferTransaction extends BaseTransaction {
 		try {
 			validateAddress(this.recipientId);
 		} catch (error) {
-			errors.push(new TransactionError(error.message, this.id, '.recipientId'));
+			errors.push(
+				new TransactionError(
+					error.message,
+					this.id,
+					'.recipientId',
+					this.recipientId,
+				),
+			);
 		}
 
 		if (
 			this.recipientPublicKey &&
 			this.recipientId !== getAddressFromPublicKey(this.recipientPublicKey)
 		) {
-			errors.push(
-				new TransactionError(
-					'recipientId does not match recipientPublicKey.',
-					this.id,
-					'.recipientId',
-				),
+			const calculatedAddress = getAddressFromPublicKey(
```

Comment:
maybe we can remove `this.recipientId !== getAddressFromPublicKey(this.recipientPublicKey)` check in above if statement since we are again checking it below.